cypress-io/cypress
# chore: add burn in mismatch error

### Additional details

Burn-in will follow the same rule as Auto Cancellation where we will not allow a user to update the config for any run already in-progress (i.e. on a different group on the same run)

Whenever trying to record a run group with a different burn-in configuration from the first group in that run, we should show an error from test runner and not permit this run group to be added to the run

### Steps to test
- Check out the branch `mabel/CYCLOUD-1704-burn-in-mismatch-error` from the cloud and run it.
- For the cypress app: pull the `mabel/chore/add-burn-in-mismatch` branch from the cypress repo
- In the project you want to use for testing, play with the configurations for `experimentalBurnIn` in your `cypress.config.js`:
``` 
experimentalBurnIn: {
    default: 1,
    flaky: 2
  }
```
```
  experimentalBurnIn: true
```
```
  experimentalBurnIn: false
```
- In the root for the cypress repo with the `mabel/chore/add-burn-in-mismatch` checked out, create a run with two or more groups and give each run a different `experimentalBurnIn` configuration. For example:
To run the first group:
```
CYPRESS_INTERNAL_ENV=development yarn cypress:run --record --key {your_project_record_key} --project ~/Documents/flaky-e2e --group group1 --ci-build-id {your_next_build_number}
```
`~/Documents/flaky-e2e` is the path to the project I use for testing

To run the second group, you can play with different experimentalBurnIn configurations:

1)
```
CYPRESS_INTERNAL_ENV=development yarn cypress:run --record --key {your_project_record_key} --project ~/Documents/flaky-e2e --group group1 --ci-build-id {your_next_build_number} --config experimentalBurnIn='{"default": 1, "flaky": 2}
```
2)
```
CYPRESS_INTERNAL_ENV=development yarn cypress:run --record --key {your_project_record_key} --project ~/Documents/flaky-e2e --group group1 --ci-build-id {your_next_build_number} --config experimentalBurnIn=true
```
3)
```
CYPRESS_INTERNAL_ENV=development yarn cypress:run --record --key {your_project_record_key} --project ~/Documents/flaky-e2e --group group1 --ci-build-id {your_next_build_number} --config experimentalBurnIn=false
```
If the `experimentalBurnIn` configuration from the first group is different from the one in the second group, you should get an error when trying to run your second group 
![image](https://github.com/cypress-io/cypress/assets/9426263/a36d4c6e-4781-4dea-9998-653adaeadb1d)



### How has the user experience changed?


### PR Tasks
